app/controllers/samples_controller.rb

```diff
@@ -947,9 +947,15 @@ def unidentified_fasta
   end
 
   def raw_results_folder
-    @file_list = @sample.results_folder_files
-    @file_path = "#{@sample.sample_path}/results/"
-    render template: "samples/raw_folder"
+    if current_user.id == @sample.user_id
+      @file_list = @sample.results_folder_files
+      @file_path = "#{@sample.sample_path}/results/"
+      render template: "samples/raw_folder"
+    else
+      render json: {
+        message: "Only the original uploader can view the sample's raw results folder."
```

Comment:
how will this be communicated to the user? or should the never have a link to the raw results if they are not allowed to see them?

Reply:
nevermind I see